pkg/diagnose/cni.go

```diff
@@ -62,6 +62,11 @@ func CNIConfig(clusterInfo *cluster.Info, _ string, status reporter.Interface) e
 	status.Start("Checking Submariner support for the CNI network plugin")
 	defer status.End()
 
+	if clusterInfo.Submariner.Status.NetworkPlugin == "" {
+		status.Warning("The network plugin hasn't been determined yet.")
```

Comment:
This implies it would be determined at some point? But shouldn't L85 below handle this case?

Reply:
Yes, it is determined later. Line 85 handles the case but produces a confusing message, indicating that the plugin isn’t supported when the reality is that Submariner doesn’t know what the plugin is yet.

Reply:
That's L80. L8(5|6) says "Submariner could not detect the CNI network plugin and is using (%q) plugin.". Isn't that the message we want? Perhaps we just need to move L85-88 here?

Reply:
I feel the message `The network plugin hasn't been determined yet` could give an impression to the user that if they retry the diagnose command after a while, the CNI plugin could be determined by then.

Reply:
@sridhargaddam yes, that’s exactly what should happen. The situation handled here only happens if `subctl diagnose` is run before the plugin has been identified; once it has, `diagnose` will show the type (or “generic” if it isn’t known).

Reply:
got it, thanks for the clarification.